### PR TITLE
Add e2e measurement for training

### PR DESCRIPTION
**Pull request by pengwa**
**Description**: add e2e measurement

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/orttraining/models/runner/training_runner.cc [hunk 1]
```diff
@@ -401,13 +401,11 @@ Status TrainingRunner::PrepareFeedNamesAndFeeds(const SessionMode mode,
     ORT_ENFORCE(params_.pipeline_parallel_size > 1);
     feed_names.push_back(pipeline_context_.forward_waited_event_name);
     OrtValue event_id;
-    const int64_t id = (mode == EvaluateStep) ? -1 : pipeline_schedule_.GetForwardWaitedEventId(
-      pipeline_context_.pipeline_stage_id,
-      static_cast<int>(step_) % pipeline_context_.num_pipeline_batches);
+    const int64_t id = (mode == EvaluateStep) ? -1 : pipeline_schedule_.GetForwardWaitedEventId(pipeline_context_.pipeline_stage_id, static_cast<int>(step_) % pipeline_context_.num_pipeline_batches);
```
Comment (ytaous) [inline orttraining/orttraining/models/runner/training_runner.cc:404]: line is too long, I thought it's recommended 120 char per line

File: orttraining/orttraining/models/runner/training_runner.cc [hunk 2]
```diff
@@ -692,25 +676,26 @@ Status TrainingRunner::RunWithoutUpdate(VectorString& feed_names,
 
   // Async launch of a session.
   pipeline_worker_pool_.workers[worker_id] = std::thread([&](
-      const size_t worker_id, const size_t step) {
+                                                             const size_t worker_id, const size_t step) {
```
Comment (ytaous) [inline orttraining/orttraining/models/runner/training_runner.cc:679]: is this space shift intentional?

File: orttraining/orttraining/models/runner/training_runner.cc [hunk 3]
```diff
@@ -692,25 +676,26 @@ Status TrainingRunner::RunWithoutUpdate(VectorString& feed_names,
...
-      pipeline_worker_pool_.worker_states[worker_id].feeds,
-      pipeline_worker_pool_.worker_states[worker_id].fetch_names,
-      &(pipeline_worker_pool_.worker_states[worker_id].fetches)) == Status::OK());
-  }, worker_id, step_);
+                    run_options,
+                    pipeline_worker_pool_.worker_states[worker_id].feed_names,
+                    pipeline_worker_pool_.worker_states[worker_id].feeds,
+                    pipeline_worker_pool_.worker_states[worker_id].fetch_names,
+                    &(pipeline_worker_pool_.worker_states[worker_id].fetches)) == Status::OK());
+  },
```
Comment (ytaous) [inline orttraining/orttraining/models/runner/training_runner.cc:697]: is this space shift intentional?

File: orttraining/orttraining/models/runner/training_runner.cc [hunk 4]
```diff
@@ -895,7 +898,8 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
             << "Total Running Time: " << total_time << " Seconds \n"
             << "Average Running Time Per Batch: " << avg_time_per_batch << " ms\n"
             << "Throughput: " << throughput << " Examples / Second\n"
```
Comment (ytaous) [inline orttraining/orttraining/models/runner/training_runner.cc:900]: Now we have 3 throughput - should we put comments on top what's the different among them?
Also, I am populating 2 throughputs to MySQL DB->PowerBI report (below), should this new entry (e2e) be added as well?

File: orttraining/orttraining/models/runner/training_runner.cc (hunk 1 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/models/runner/training_runner.cc:404]: +1

File: orttraining/orttraining/models/runner/training_runner.cc (hunk 4 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/models/runner/training_runner.cc:900]: use the EndToEnd Throughput.

Comment (ytaous) [reply orttraining/orttraining/models/runner/training_runner.cc:900]: so discard the other 2 TP (for perf monitor), just use e2e TP then (after this is merged)? 

---
In reply to: [431491842](https://github.com/microsoft/onnxruntime/pull/4049#discussion_r431491842) [](ancestors = 431491842)

Comment (ytaous) [reply orttraining/orttraining/models/runner/training_runner.cc:900]: BTY - the bert perf test only run 100 steps - looks like we need to extend it to 200 so it's > 128 

---
In reply to: [431522622](https://github.com/microsoft/onnxruntime/pull/4049#discussion_r431522622) [](ancestors = 431522622,431491842)

File: orttraining/orttraining/models/runner/training_runner.cc (hunk 2 above)
Comment (pengwa) [reply orttraining/orttraining/models/runner/training_runner.cc:679]: it is done automatically by my VS clang formator.

File: orttraining/orttraining/models/runner/training_runner.cc (hunk 4 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/models/runner/training_runner.cc:900]: right... we can have it run longer, say 512+128 steps

Comment (ytaous) [reply orttraining/orttraining/models/runner/training_runner.cc:900]: sure, will do the same for gpt-2 then

**Comment by pengwa** [comment]
thank you @SherlockNoMad  @ytaous

**Pull request merged**